### PR TITLE
fix gh-4328 This was done for python3, but not python2.

### PR DESCRIPTION
**Pull request by nouiz**
fix gh-4328

### CODE REVIEW

File: theano/tensor/opt.py [hunk 1]
```diff
@@ -1640,6 +1640,8 @@ def local_fill_sink(node):
             if not r:
                 continue
             replacements.update(r)
+    for v in replacements.values():
+        v.tag.values_eq_approx = values_eq_approx_remove_nan
```
Comment (abergeron) [inline theano/tensor/opt.py]: Why should we ignore nans in all cases here.

File: theano/tensor/opt.py [hunk 2]
```diff
@@ -3849,9 +3851,11 @@ def local_merge_switch_same_cond(node):
     if not all(s.owner.inputs[0] is cond for s in node.inputs[1:]):
         return
     # pull out switch
-    return [T.switch(cond,
-                     node.op(*[s.owner.inputs[1] for s in node.inputs]),
-                     node.op(*[s.owner.inputs[2] for s in node.inputs]))]
+    ret = T.switch(cond,
+                   node.op(*[s.owner.inputs[1] for s in node.inputs]),
+                   node.op(*[s.owner.inputs[2] for s in node.inputs]))
+    ret.tag.values_eq_approx = values_eq_approx_remove_nan
```
Comment (abergeron) [inline theano/tensor/opt.py]: Again why do we ignore nans here?

File: theano/tensor/opt.py [hunk 3]
```diff
@@ -5304,7 +5308,9 @@ def local_zero_div(node):
     if isinstance(node.op, T.Elemwise) and isinstance(
             node.op.scalar_op, (theano.scalar.IntDiv, theano.scalar.TrueDiv)):
         if local_mul_canonizer.get_constant(node.inputs[0]) == 0:
-            return [broadcast_like(0, node.outputs[0], node.fgraph)]
+            ret = broadcast_like(0, node.outputs[0], node.fgraph)
+            ret.tag.values_eq_approx = values_eq_approx_remove_nan
```
Comment (abergeron) [inline theano/tensor/opt.py]: Again ...

File: theano/tensor/opt.py (hunk 1 above)
Comment (nouiz) [inline theano/tensor/opt.py]: I didn't understand well, but all those optimizations are removing nan and so without this, it make the daily buildbot fail.

Comment (lamblin) [inline theano/tensor/opt.py]: I'm pretty sure this optimization should not remove NaNs.

File: theano/tensor/opt.py (hunk 2 above)
Comment (lamblin) [inline theano/tensor/opt.py]: Not this one either.

File: theano/tensor/opt.py (hunk 3 above)
Comment (lamblin) [inline theano/tensor/opt.py]: I agree for that one: it is replacing `0/x` by `0`, so when evaluated at `x == 0`, a NaN is removed.
I think fixing this one should be enough to fix the test, and the ones in `local_merge_switch_same_cond`, and probably in `local_fill_sink`, should be removed.

**Comment by nouiz** [comment]
I just updated this PR by removing the change @lamblin didn't agree. This still change one opt, maybe it was the root cause. No time to make sure it fix all the error I wanted to fix in the daily buildot.

I think we can merge this PR.

**Pull request merged**